src/Transition/CreateNewIDFUsingRulesV8_8_0.f90

```diff
@@ -366,6 +389,66 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                  OutArgs(15:CurArgs+1)=InArgs(14:CurArgs)  !
                  CurArgs = CurArgs + 1
 
+             CASE ('BUILDINGSURFACE:DETAILED')
+                 CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                 nodiff=.false.
+                 IF ( SameString(InArgs(5), 'Foundation' ) .and. SameString(InArgs(2), 'Floor' ) ) THEN
+                   NumPerimObjs = GetNumObjectsFound('SURFACEPROPERTY:EXPOSEDFOUNDATIONPERIMETER')
+                   PerimNum = 0
+                   DO xcount=1,NumPerimObjs
+                     CALL GetObjectItem('SURFACEPROPERTY:EXPOSEDFOUNDATIONPERIMETER',xcount,Alphas,NumAlphas,Numbers,NumNumbers,Status)
+                     IF ( SameString( TRIM(Alphas(1)), TRIM(InArgs(1)) ) ) THEN
+                       PerimNum = xcount
```

Comment:
Could break here once one is found.  Also, could have made it a boolean for a little better clarity since I think you aren't actually using the integer value except as a flag.  No worries though.